File: sistemaBackHerrera/src/main/java/com/sistemabackherrera/model/Automovil.java
package com.sistemabackherrera.model;


import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.math.BigDecimal;

@Entity
@Table(name="tb_automovil")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Automovil {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne()
    @JoinColumn(name="id_marca")
    private Marca marca;

    @OneToOne()
    @JoinColumn(name="id_motor")
    private Motor motor;

    @OneToOne()
    @JoinColumn(name="id_color")
    private Color color;

    @Column(nullable = false)
    private Long anioFabricacion;

    @Column(nullable = false)
    private BigDecimal cilindrada;

    @Column(nullable = false)
    private Long numPuertas;

    @Column(nullable = false)
    private BigDecimal precio;

}

File: sistemaFrontHerrera/src/app/modelos/Automovil.ts
import { Color } from "./Color";
import { Marca } from "./Marca";
import { Motor } from "./Motor";

export class Automovil{
   id:number;
   marca!:Marca;
   motor!:Motor;
   color!:Color;
   anioFabricacion:number;
   cilindrada:number;
   numPuertas:number;
   precio:number;

   constructor(){
    this.id=0;
    this.anioFabricacion=0;
    this.cilindrada=0;
    this.numPuertas=0;
    this.precio=0;
   }
}